src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTextAreaUI.java

```diff
@@ -138,7 +138,8 @@ public Dimension getMinimumSize(JComponent c) {
 
     /**
      * Creates the view for an element.  Returns a WrappedPlainView or
-     * PlainView.
+     * PlainView if non-bidi text is handled
+     * else returns PlainParagraph or BoxView if bidi text is being handled.
```

Comment:
You cannot specify that this method returns `PlainParagraph`, it's a private class. I think we should relax the specification to allow other types to be returned without mentioning the exact types.

This issue is also relevant to `BasicTextFieldUI`, yet [it does not state explicitly](https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/javax/swing/plaf/basic/BasicTextFieldUI.html#create(javax.swing.text.Element)) what types are valid, it only gives a hint that a `FieldView` is returned:

> Creates a view (FieldView) based on an element.

So, I propose the following text: *“Creates a view for an element. Usually returns a WrappedPlainView or PlainView.”*

The `@return` tag may be changed to “a view (for the the element)” for this class as well as for `BasicTextFieldUI`.

Reply:
I guess we can mention "Creates a view for an element." only.
`Usually returns a WrappedPlainView.. ` can give problem to TCK team to create testcase for all scenario. Either we mention the full `returns WrappedView or PlainView. or ParagraphView or BoxView`

Reply:
> I guess we can mention "Creates a view for an element." only. `Usually returns a WrappedPlainView.. ` can give problem to TCK team to create testcase for all scenario.

Then the current code shouldn't have passed the TCK as the spec requires either `WrappedView` or `PlainView`.

> Either we mention the full `returns WrappedView or PlainView. or ParagraphView or BoxView`

We can't specify it returns a package-private class.